aptos-move/framework/src/extended_checks.rs

```diff
@@ -479,8 +489,59 @@ impl<'a> ExtendedChecker<'a> {
                 .fun_attributes
                 .entry(fun.get_simple_name_string().to_string())
                 .or_default()
-                .push(KnownAttribute::randomness());
+                .push(KnownAttribute::randomness(randomness_annotation.max_gas));
+        }
+    }
+
+    fn get_randomness_max_gas_declaration(
+        &self,
+        fun: &FunctionEnv,
+    ) -> Result<Option<RandomnessAnnotation>, String> {
+        for attr in fun.get_attributes() {
+            let Attribute::Apply(_, name, sub_attrs) = attr else {
+                continue;
+            };
+            if self.env.symbol_pool().string(*name).as_str() != RANDOMNESS_ATTRIBUTE {
+                continue;
+            }
+            if sub_attrs.len() >= 2 {
+                return Err(
+                    "Randomness attribute should only have one `max_gas` property.".to_string(),
+                );
+            }
+            if let Some(sub_attr) = sub_attrs.first() {
+                let Attribute::Assign(_, key_symbol, attr_val) = sub_attr else {
+                    return Err(
+                        "Randomness attribute should only have one `max_gas` property.".to_string(),
+                    );
+                };
+                let key = self.env.symbol_pool().string(*key_symbol);
+                if key.as_str() != RANDOMNESS_MAX_GAS_CLAIM {
+                    return Err(format!(
+                        "Unknown key for randomness attribute: {}",
+                        key.as_str()
+                    ));
+                }
+
+                let msg =
+                    "Property `max_gas` should be a positive integer less than 2^64".to_string();
+                let AttributeValue::Value(_, Value::Number(val)) = attr_val else {
+                    return Err(msg);
+                };
+                if val.abs() != *val {
+                    return Err(msg);
+                }
+                let (_, digits) = val.to_u64_digits();
+                if digits.len() >= 2 {
+                    return Err(msg);
+                }
+                let max_gas_declaration_seen = Some(digits.first().copied().unwrap_or(0));
```

Comment:
shouldn't it be `unwrap_or(RandomnessAnnotation::default())` ?

Reply:
Then add some comments?

Reply:
fixed